src/services/file.forked.services.js

```diff
@@ -19,14 +24,25 @@ import {
  */
 const getFileFromForkedRepoByPath = async (owner, repo, branch, path) => {
   try {
-    const { data } = await axios.get(`${GITHUB_APP_BASE_URL}/files`, {
-      params: {
-        owner: owner,
-        repo: repo,
-        branch: branch,
-        path: path,
-      },
-    });
+    if(!GITHUB_APP_BASE_URL || GITHUB_APP_BASE_URL.trim() === '') {
+      throw new MissingGitHubAppDomainError();
+    }
+    if(!CHANGELOG_PR_BRIDGE_SECRET_KEY || CHANGELOG_PR_BRIDGE_SECRET_KEY.trim() === '') {
+      throw new MissingChangelogBridgeSecretKeyError();
+    }
```

Comment:
This check is being done before any call to the github app. Cant you move this to the utils folder and clean this up? The same goes for error handling. You can do something like this:

```js
const handleErrors = (errorContext, error, owner, branch) => {
  const errorMessage = error.response?.data?.error?.message || error.message;
  console.error(
    `${errorContext}: ${owner}/${branch}:`,
    errorMessage
  );
  if (errorMessage.includes("GitHub App")) {
    throw new GitHubAppSuspendedOrNotInstalledError();
  }
  throw new CreateOrUpdateContentError();
};

const validateEnvVariables = () => {
  if(!GITHUB_APP_BASE_URL || GITHUB_APP_BASE_URL.trim() === '') {
    throw new MissingGitHubAppDomainError();
  }
  if(!CHANGELOG_PR_BRIDGE_SECRET_KEY || CHANGELOG_PR_BRIDGE_SECRET_KEY.trim() === '') {
    throw new MissingChangelogBridgeSecretKeyError();
  }
};
```

Reply:
Make sense,

We should put all validator logic under the `utils` folder under the `validators.utils.js` file. I will take a look to that.

Reply:
Changes done @ashwin-pc swhin. The following approach was followed:

```js
// validators.utils.js
export const checkGithubAppDomainIsAvailable = () => {
  if (
    !CHANGELOG_PR_BRIDGE_URL_DOMAIN ||
    CHANGELOG_PR_BRIDGE_URL_DOMAIN.trim() === ""
  ) {
    throw new MissingChangelogPrBridgeUrlDomainError();
  }
};

export const checkChangelogPrBridgeApiKeyIsAvailable = () => {
  if (
    !CHANGELOG_PR_BRIDGE_API_KEY ||
    CHANGELOG_PR_BRIDGE_API_KEY.trim() === ""
  ) {
    throw new MissingChangelogPrBridgeApiKeyError();
  }
};
```

```js
// handlers.utils.js
import {
  GetContentError,
  CreateOrUpdateContentError,
  DeleteContentError,
  GitHubAppSuspendedOrNotInstalledError,
  UnauthorizedRequestToPRBridgeServiceError

} from "../errors/index.js";

export const handleChangelogPRBridgeResponseErrors = (
  error,
  crudOperation,
) => {

  switch (error.status) {
    case 404:
      console.error(`File '${path}' not found.`);
      return;
    case 401:
      throw new UnauthorizedRequestToPRBridgeServiceError();
    case 403:
      throw new GitHubAppSuspendedOrNotInstalledError();
    default:
      if (crudOperation === "READ") {
        throw new GetContentError();
      }
      else if (crudOperation === "CREATE_OR_UPDATE") {
        throw new CreateOrUpdateContentError();
      }
      else if (crudOperation === "DELETE"){
        throw new DeleteContentError();
      }
      else {
        throw error;
      }
  }
};
```